./scripts_regression_tests.py with compiler option needs refinement

We need a minor change in scripts_regression_tests.py so that this test will pass with the --compiler option.
```
======================================================================
FAIL: test_cime_case_test_custom_project (__main__.K_TestCimeCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 1715, in test_cime_case_test_custom_project
    env_changes="unset CIME_GLOBAL_WALLTIME &&")
  File "./scripts_regression_tests.py", line 821, in _create_test
    expected_stat=expected_stat)
  File "./scripts_regression_tests.py", line 60, in run_cmd_assert_result
    test_obj.assertEqual(stat, expected_stat, msg=msg)
AssertionError: 
COMMAND: unset CIME_GLOBAL_WALLTIME && /home/jedwards/cesm2_0_alpha/cime/scripts/create_test --no-setup --machine=melvin --project=testproj ERS_P1.f19_g16_rx1.A -t fake_testing_only_20180106_125308 --baseline-root /scratch/cluster/jedwards/scripts_regression_test.20180106_124431/baselines --compiler=nag --test-root=/scratch/cluster/jedwards/scripts_regression_test.20180106_124431 --output-root=/scratch/cluster/jedwards/scripts_regression_test.20180106_124431
FROM_DIR: /home/jedwards/cesm2_0_alpha/cime/scripts/tests
SHOULD HAVE WORKED, INSTEAD GOT STAT 1
OUTPUT: Testnames: ['ERS_P1.f19_g16_rx1.A.melvin_nag']
ERRPUT: ERROR: Compiler nag not valid for machine melvin
```

--- a/scripts/tests/scripts_regression_tests.py
+++ b/scripts/tests/scripts_regression_tests.py
@@ -804,9 +804,9 @@ class TestCreateTestCommon(unittest.TestCase):
         extra_args.append("--baseline-root {}".format(self._baseline_area))
         if NO_BATCH:
             extra_args.append("--no-batch")
-        if TEST_COMPILER:
+        if TEST_COMPILER and ([extra_arg for extra_arg in extra_args if "--compiler" in extra_arg] == []):
             extra_args.append("--compiler={}".format(TEST_COMPILER))
-        if TEST_MPILIB:
+        if TEST_MPILIB and ([extra_arg for extra_arg in extra_args if "--mpilib" in extra_arg] == []):
             extra_args.append("--mpilib={}".format(TEST_MPILIB))
         extra_args.append("--test-root={0} --output-root={0}".format(TEST_ROOT))
 
@@ -1711,7 +1711,8 @@ class K_TestCimeCase(TestCreateTestCommon):
     ###########################################################################
         test_name = "ERS_P1.f19_g16_rx1.A"
         machine, compiler = "melvin", "gnu" # have to use a machine both models know and one that doesn't put PROJECT in any key paths
-        self._create_test(["--no-setup", "--machine={}".format(machine), "--project=testproj", test_name], test_id=self._baseline_name,
+        self._create_test(["--no-setup", "--machine={}".format(machine), "--compiler={}".format(compiler), "--project=testproj", test_name],
+                          test_id=self._baseline_name,
                           env_changes="unset CIME_GLOBAL_WALLTIME &&")
 
         casedir = os.path.join(self._testroot,

